pkg/apis/servicecatalog/types.go

```diff
@@ -270,9 +270,17 @@ type InstanceSpec struct {
 	// provisioned from.
 	PlanName string
 
-	// Parameters is a YAML representation of the properties to be
+	// List of sources to populate parameters.
+	// When a key exists in multiple
+	// sources, the value associated with the last source will take precedence.
+	// Values defined by a Parameter with a duplicate key will take precedence.
+	// +optional
+	ParametersFrom []ParametersFromSource
+
+	// Parameters is a set of the parameters to be
 	// passed to the underlying broker.
-	Parameters *runtime.RawExtension
+	// +optional
+	Parameters []Parameter
```

Comment:
Based on our discussion in the Aug 1 design meeting, I would have expected that this PR would only do the following:

1.  Add a note to the existing `Parameters` field that it is deprecated and will be removed in the future
2.  Add the new `ParametersFrom` field, with the `Value` and `SecretKeyRef` types only
3.  Add a validation that prohibits both `Parameters` and `ParametersFrom` from being set
4.  Document the merging behaviors as we discussed in the Aug 1 call

Perhaps we had a miscommunication?